packages/blade/src/components/Input/BaseInput/BaseInput.tsx

```diff
@@ -233,7 +239,21 @@ export type BaseInputProps = FormInputLabelProps &
      * sets the autocapitalize behavior for the input
      */
     autoCapitalize?: 'none' | 'sentences' | 'words' | 'characters';
-  } & TestID;
+  } & TestID &
+  Platform.Select<{
+    native: {
+      /**
+       * The callback function to be invoked when the value of the input field is submitted.
+       */
+      onSubmit?: FormInputOnEvent;
+    };
+    web: {
+      /**
+       * This is a react-native only prop and has no effect on web.
+       */
+      onSubmit?: undefined;
```

Comment:
Is there any way to hide this from autocomplete itself 🤔 ?

Reply:
That would be quite chaotic since we reuse BaseInput in TextInput and other components where we do pass onSubmit, there its value is a function or undefined, if we completely remove it as a prop from web then it will not even accept undefined as a valid value